client/ayon_core/tools/loader/ui/product_group_dialog.py

```diff
@@ -32,11 +42,46 @@ def __init__(self, controller, parent):
         self._controller = controller
         self._group_btn = group_btn
         self._group_name_input = group_name_input
+        self._group_picker_btn = group_picker_btn
+        self._group_picker_menu = group_picker_menu
+        self._group_picker_menu.triggered.connect(self._on_picker_clicked)
 
     def set_product_ids(self, project_name, product_ids):
         self._project_name = project_name
         self._product_ids = product_ids
 
+        # Get parent folders, then for all products under those folders
+        # get all product groups so we can provide those as 'predefined'
+        # entries
+        folder_ids = {
+            product["folderId"] for product in
+            ayon_api.get_products(
+                project_name, product_ids=product_ids, fields={"folderId"})
+        }
+        product_groups = set()
+        for product in ayon_api.get_products(
```

Comment:
Please do all backend logic via controller, not in UI. This should be implemented in `tools/loader/models/products.py` and accessible via controller logic.

Reply:
Perfect - thank you. I was hoping you could point me to where I should move this.

Reply:
I would recommend to just implement
```
def get_group_names_for_product_ids(self, project_name, product_ids):
    group_names = set()
    for product_id in product_ids
        product_item = self.get_product_item(project_name, product_id)
        if product_item is not None:
           group_names.add(product_item.group_name)
    group_names.discard(None)
    return group_names
```

Reply:
Correct - I want to show all product groups that are under the "selected folders" (there may be multiple) which I'm currently doing through logic retrieved via the controller. Are you saying to just make that logic more embedded inside the controller so it's "one call away" instead of three by implementing e.g. this on the controller?

```python
def get_current_folders_product_group_names(self):
    folder_ids = self._controller.get_selected_folder_ids()
    product_items = self._controller.get_product_items(
        project_name=self._controller.get_selected_project_name(),
        folder_ids=folder_ids)
    product_groups = {
            product_item.group_name for product_item in product_items
    }
    product_groups.discard(None)
    return product_groups
```

Reply:
> Are you saying to just make that logic more embedded inside the controller so it's "one call away" instead of three by implementing e.g. this on the controller?

Yes, and also avoid using `get_selected_folder_ids`.

Reply:
Moved more logic into the controller with https://github.com/ynput/ayon-core/pull/668/commits/587665b106eef1e116a549b958667b6323ae0bff

Is that what you meant?

Reply:
Bump - just seeing if I still need to change anything. :)

Reply:
Marking resolved - let me know if still an issue.